src/app/course/course-challenge-snippet/course-challenge-snippet.component.html

```diff
@@ -0,0 +1,104 @@
+<div class="challengePage">
+    <tui-accordion class="accordionContainer" [closeOthers]="false">
+        <tui-accordion-item class="title" open="true" [disableHover]="true">
+            Upcoming Challenges
+            <ng-template tuiAccordionItemContent>
+                <div *ngFor="let challenge of upcomingEvents" class="challenge">
+                    <div class="challengeType"> Challenge </div>
+                    <div class="name">{{ challenge.name }}</div>
+
+                    <div class="dates">
+                        <div>From: {{ challenge.start_date | date: 'EEEE, MMM d, y, h:mm a' }}</div>
+                        <div>Until: {{ challenge.end_date | date: 'EEEE, MMM d, y, h:mm a' }}</div>
+                    </div>
+                    <div class="availability">
+                        <tui-tag *ngIf="challenge.is_open; else challengeComingSoon" size="m" status="success" value="Available"></tui-tag>
+                        <ng-template #challengeComingSoon>
+                            <tui-tag size="m" status="error" value="Coming Soon"></tui-tag>
+                        </ng-template>
+                    </div>
+                    <div class="grade">
+                        <ng-container *ngIf="challenge.is_not_available_yet ; else totalEventGrade">
+                            Grade: TBA
+                        </ng-container>
+                        <ng-template #totalEventGrade>
+                            Grade: {{ challenge.total_event_grade | number: '1.0-1' || 'No Grade'}}%
```

Comment:
What is the difference between 0% and No Grade?
Under what condition will it display No Grade?